slice-mapper/src/main/java/com/cognifide/slice/mapper/GenericSlingMapper.java

```diff
@@ -140,25 +140,28 @@
 	 */
 	private <T> T mapResourceToObject(final Resource resource, final T object) {
 		ValueMap valueMap = resource.adaptTo(ValueMap.class);
-		try {
-			Class<?> type = object.getClass();
-			Field[] fields = ReflectionHelper.readAllDeclaredFields(type);
-			for (Field field : fields) {
-				MapperStrategy mapperStrategy = mapperStrategyFactory.getMapperStrategy(field
-						.getDeclaringClass());
-				if (shouldFieldBeMapped(field, mapperStrategy)) {
-					Object value = mapResourceToField(resource, valueMap, field);
-					FieldUtils.writeField(field, object, value, ReflectionHelper.FORCE_ACCESS);
+		if (valueMap != null) {
```

Comment:
We should not change a contract of FieldProcessor this way. FieldProcessor can currently expect null values under ValueMap parameter. Therefore, I'd suggest to add appropriate null checks to processors and update documentation appropriately